### PR TITLE
Hold mutex lock shorter when processing inter-pod affinity/anti-affin…

### PR DESCRIPTION
**Pull request by bsalamat**
…ity priority function

**What type of PR is this?**
> Uncomment only one, leave it on its own line:
>
> /kind api-change
> /kind bug

/kind cleanup

> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:
This PR improves parallelism and potentially increases performance when processing the inter-pod affinity/anti-affinity priority function.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

/sig scheduling

### CODE REVIEW

File: pkg/scheduler/algorithm/priorities/interpod_affinity.go [hunk 1]
```diff
@@ -92,15 +92,15 @@ func (p *podAffinityPriorityMap) processTerm(term *v1.PodAffinityTerm, podDefini
...
 	match := priorityutil.PodMatchesTermsNamespaceAndSelector(podToCheck, namespaces, selector)
 	if match {
-		func() {
-			p.Lock()
-			defer p.Unlock()
-			for _, node := range p.nodes {
-				if priorityutil.NodesHaveSameTopologyKey(node, fixedNode, term.TopologyKey) {
+		for _, node := range p.nodes {
+			if priorityutil.NodesHaveSameTopologyKey(node, fixedNode, term.TopologyKey) {
+				func() {
```
[the remark this answers is not in the document]
Comment (bsalamat) [reply pkg/scheduler/algorithm/priorities/interpod_affinity.go]: Yes, it is just to ensure unlocking. In this case that the body is trivial, I agree that your version is fine.

File: pkg/scheduler/algorithm/priorities/interpod_affinity.go [hunk 2]
```diff
@@ -92,15 +92,13 @@ func (p *podAffinityPriorityMap) processTerm(term *v1.PodAffinityTerm, podDefini
...
-			defer p.Unlock()
-			for _, node := range p.nodes {
-				if priorityutil.NodesHaveSameTopologyKey(node, fixedNode, term.TopologyKey) {
-					p.counts[node.Name] += weight
-				}
+		for _, node := range p.nodes {
+			if priorityutil.NodesHaveSameTopologyKey(node, fixedNode, term.TopologyKey) {
+				p.Lock()
+				p.counts[node.Name] += weight
+				p.Unlock()
```
Comment (krmayankk) [inline pkg/scheduler/algorithm/priorities/interpod_affinity.go]: What other processes call this lock ?  Curious if this actually causes some issues or just an optimization ?

**Comment by krmayankk** [comment]
@bsalamat @Huang-Wei curious did my lgtm caused it to be merged since it was already approved ?  Were you waiting for something. May be for  PR from maintainers , we should always put a hold or say WIP :-)

**Pull request merged**